src/hotspot/share/opto/loopopts.cpp

```diff
@@ -4209,6 +4210,34 @@ bool PhaseIdealLoop::duplicate_loop_backedge(IdealLoopTree *loop, Node_List &old
   return true;
 }
 
+bool PhaseIdealLoop::autovectorize(IdealLoopTree* lpt, ResourceArea* arena) {
+  // Counted loop only
+  if (!lpt->is_counted()) { return false; }
+  CountedLoopNode* cl = lpt->_head->as_CountedLoop();
+
+  // Main-loop only
+  if (!cl->is_main_loop()) { return false; }
+
+  VLoop vloop(lpt, false);
+  if (vloop.check_preconditions()) {
+    // Ensure that all data structures from autovectorization are deallocated later.
+    ResourceMark rm(arena);
+    SuperWord sw(arena, vloop);
+    if (sw.transform_loop()) {
+      return true;
+    }
+  }
+
+  // This counted main-loop either failed preconditions,
+  // or in SuperWord. From now on only unroll the loop.
+  if (cl->has_passed_slp()) {
+    C->set_major_progress();
+    cl->set_notpassed_slp();
+    cl->mark_do_unroll_only();
+  }
```

Comment:
This is not related to vectorization. Can you move it back and execute based on return value from `autovectorize()` which is not used now?